test/test-users.js

```diff
@@ -0,0 +1,402 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const spies = require('chai-spies');
+const mongoose = require('mongoose');
+const app = require('../server').app;
+
+const {
+  User
+} = require('../users/models');
+const should = chai.should();
+
+var UrlPattern = require('url-pattern');
+const pattern = {
+  "/users": new UrlPattern('/users'),
+  "/users/:userId": new UrlPattern('/users/:userId')
+}
+
+chai.use(chaiHttp);
+chai.use(spies);
+
+describe('Test /user endpoints with empty database', () => {
+  beforeEach((done) => {
+    // Approach #1 Drop database and `ensureIndexes()`
+    mongoose.connection.db.dropDatabase()
+      .then(() => User.ensureIndexes())
+      .then(() => done());
+
+    // Approach #2 Remove items, retains indexes
+    // User.remove({}).then( () => done())
+
+  });
+  describe('GET /users', () => {
+    it('should return an empty list of users initially', () => {
+      return chai.request(app)
+        .get('/users')
+        .then(res => {
+          // Check that it's an empty array
+          res.should.have.status(200);
+          res.type.should.equal('application/json');
+          res.charset.should.equal('utf-8');
+          res.body.should.be.an('array');
+          res.body.length.should.equal(0);
+        });
+    });
+  });
+});
+
+describe('Test /user endpoints with prepopulated users', () => {
+  // config dummy users to be used throughout across the tests
+  let johnDoe = {
+    username: 'john.doe@example.com',
+    password: 'letmein',
+    firstName: 'John',
+    lastName: 'Doe'
+  };
+  let janeDoe = {
+    username: 'jane.doe@example.com',
+    password: 'password',
+    firstName: 'Jane',
+    lastName: 'Doe'
+  };
+
+  beforeEach((done) => {
+    // Approach #2.5 Remove items and then insert dummy users
+    User.remove({})
```

Comment:
Ah okay, so in that case, we don't really need an `afterEach` routine. Yeah, that makes sense. Maybe not so important in this current app cause just one model, but would this evolve into this sorta thing:

```js
Promise.all([User.remove({}).exec(), OtherModel.remove({}).exec(), YetAnotherModel.remove({}).exec()])
  .then(() => seedDatabase());
```